packages/bot-web-ui/src/components/dashboard/tour-trigger-dialog.tsx

```diff
@@ -20,35 +20,54 @@ type TTourTriggrerDialog = {
 const TourTriggrerDialog = ({
     active_tab,
     is_tour_dialog_visible,
-    is_tour_ended,
+    has_tour_ended,
     setTourDialogVisibility,
     setBotBuilderTourState,
     setOnBoardTourRunState,
     setTourActive,
-    setIsTourEnded,
+    setHasTourEnded,
 }: TTourTriggrerDialog) => {
     const toggleTour = (value: boolean, type: string) => {
-        if (type === 'onConfirm') {
-            if (active_tab === 0) {
-                setTourActive(value);
-                setOnBoardTourRunState(value);
-                setTourDialogVisibility(false);
+        if (tour_type.key === 'onboard_tour') {
+            if (type === 'onConfirm') {
+                if (active_tab === 0) {
+                    setTourActive(value);
+                    setOnBoardTourRunState(value);
+                    setTourDialogVisibility(false);
+                } else {
+                    setBotBuilderTourState(value);
+                    setTourDialogVisibility(false);
+                }
+                setHasTourEnded(value);
             } else {
                 setBotBuilderTourState(value);
                 setTourDialogVisibility(false);
             }
-            setIsTourEnded(value);
-        } else {
-            setTourSettings(new Date().getTime(), `${tour_type.key}_token`);
-            setTourDialogVisibility(false);
+            tour_type.key = 'onboard_tour';
+        } else if (tour_type.key === 'bot_builder') {
+            if (type === 'onConfirm') {
+                if (active_tab === 0) {
+                    setTourActive(value);
+                    setOnBoardTourRunState(value);
+                    setTourDialogVisibility(false);
+                } else {
+                    setBotBuilderTourState(value);
+                    setTourDialogVisibility(false);
+                }
+                setHasTourEnded(value);
+            } else {
+                setTourSettings(new Date().getTime(), 'bot_builder_token');
+                setTourDialogVisibility(false);
+            }
+            tour_type.key = 'bot_builder';
         }
```

Comment:
`setTourDialogVisibility(false);`  is present in all conditions, can we have it once on each block instead?